### ISSUE TITLE
kdwsdl2cpp creates invalid code with Qt6

### ISSUE THREAD

**Issue by jtjbrady** [KDAB/KDSoap#249]
Tested using both MSVC2019 and GCC 11.2.0 with Qt 6.3.1 on Windows.
Error occurs on latest master and KDSoap 2.0 branches.
Essentially the Qt6 version of kdwsdl2cpp is adding an empty invalid class declaration in the header file.

Compilation of kdsoap fails with:

```
FAILED: examples/bank_gui/CMakeFiles/bank_gui.dir/bank_gui_autogen/mocs_compilation.cpp.obj
C:\PROGRA~2\MICROS~4\2019\PROFES~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\cl.exe  /nologo /TP -DQT_CORE_LIB -DQT_FATAL_ASSERT -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_CAST_FROM_ASCII -DQT_NO_CAST_FROM_BYTEARRAY -DQT_NO_CAST_TO_ASCII -DQT_NO_FOREACH -DQT_WIDGETS_LIB -DQURL_NO_CAST_FROM_STRING -DUNICODE -DUSE_EXCEPTIONS -DWIN32 -DWIN64 -D_ENABLE_EXTENDED_ALIGNED_STORAGE -D_SCL_SECURE_NO_WARNINGS -D_UNICODE -D_WIN64 -ID:\kdsoap.git.Qt6\examples\bank_gui -ID:\KDSoap.git\examples\bank_gui -ID:\kdsoap.git.Qt6\examples\bank_gui\bank_gui_autogen\include -ID:\KDSoap.git\examples\..\src -ID:\KDSoap.git\examples\..\src\KDSoapClient -ID:\KDSoap.git\examples\..\src\KDSoapServer -external:IQ:\Qt\6.3.1\msvc2019_64\include\QtCore -external:IQ:\Qt\6.3.1\msvc2019_64\include -external:IQ:\Qt\6.3.1\msvc2019_64\mkspecs\win32-msvc -external:IQ:\Qt\6.3.1\msvc2019_64\include\QtNetwork -external:IQ:\Qt\6.3.1\msvc2019_64\include\QtWidgets -external:IQ:\Qt\6.3.1\msvc2019_64\include\QtGui -external:W0 /DWIN32 /D_WINDOWS /W3 /GR /EHsc /MDd /Zi /Ob0 /Od /RTC1 -Zc:__cplusplus -permissive- -utf-8 -std:c++17 /showIncludes /Foexamples\bank_gui\CMakeFiles\bank_gui.dir\bank_gui_autogen\mocs_compilation.cpp.obj /Fdexamples\bank_gui\CMakeFiles\bank_gui.dir\ /FS -c D:\kdsoap.git.Qt6\examples\bank_gui\bank_gui_autogen\mocs_compilation.cpp
D:\kdsoap.git.Qt6\examples\bank_gui\wsdl_BLZService.h(26): error C2332: 'class': missing tag name
D:\kdsoap.git.Qt6\examples\bank_gui\wsdl_BLZService.h(26): warning C4094: untagged 'class' declared no symbols
```

All the examples fail, I've just copied the output from the last one.

Diff between the output of kdwsdl2cpp compiled with Qt5 and Qt6:

```
--- kdsoap.git.Qt5/examples/bank_gui/wsdl_BLZService.h  2022-08-16 22:50:47.831284100 +0100
+++ kdsoap.git.Qt6/examples/bank_gui/wsdl_BLZService.h  2022-08-16 23:03:13.241140400 +0100
@@ -23,6 +23,7 @@
 class KDSoapPendingCallWatcher;
 class KDSoapValue;
 class TNS__DetailsType;
+class ;
 
 class TNS__DetailsType
 {
```

**Comment by jtjbrady** [comment]
It appears the problem is in libkode, specifically with the logic in void Printer::printHeader(const File &file) on lines 744, 751, 753 and 765.

744: fwdClasses += QString(); // for proper closing of the namespace blocks below
Adds a null QString to the loop below

751: const QStringList seg = fwd.split(QLatin1String("::"));
This appears to be the root of the problem, fwd.split when passed a null QString with:

- Qt5 returns a list containing a single null QString
- Qt6 returns a list containing a single empty QString

753: const QString clas = seg.isEmpty() ? QString() : seg.last();
seg can never be empty

765: if (!clas.isNull()) {
clas is empty with Qt6 not null.

**Comment by dfaure-kdab** [comment]
Hello,

This is surprising, KDSoap builds for Qt6 in our CI and on my machine.
In commit c7d27c7d I updated the libkode submodule to fix building with Qt 6.3, by including libkode commit 50138673e23c2 which looks exactly like what you describe.

Are you using kdsoap from git master *and* did you do `git submodule update`?

**Comment by jtjbrady** [comment]
Yes I did a git submodule update. What I failed to do was use the correct command to checkout the master git branch, so I did the same tests twice on kdsoap2.0, rather than kdsoap2.0 and master.  That's embarrassing.  I can confirm the master branch works fine.
